docs/components/saas/byok/cost-and-troubleshooting.md

```diff
@@ -0,0 +1,40 @@
+---
+id: cost-and-troubleshooting
+title: "Cost implications and troubleshooting"
+description: "Learn more about cost implications and troubleshooting."
+---
+
+Using external encryption keys with **Amazon KMS** incurs costs directly in your Amazon account. Camunda does not charge for the feature itself, but you are responsible for all Amazon KMS usage.
+
+## Cost implications
+
+| Cost type       | Description                                                              | Notes                                                      |
+| --------------- | ------------------------------------------------------------------------ | ---------------------------------------------------------- |
+| KMS key storage | Monthly charge for each KMS key                                          | Depends on Amazon region and key type                      |
+| API requests    | Charges for KMS API calls (Encrypt, Decrypt, GenerateDataKey, ReEncrypt) | Costs increase with frequent operations                    |
+| CloudTrail logs | Charges for storing and accessing CloudTrail events                      | Includes encryption/decryption activity by Camunda cluster |
+
+:::warning Cost responsibility
+You are responsible for monitoring Amazon KMS usage and associated costs.
+:::
+
+### Cost optimization tips
+
+- Use separate keys only when necessary to avoid extra storage fees.
+- Aggregate audit logging to reduce frequent API calls.
+- Review CloudTrail retention settings to balance compliance and storage cost.
+
+## Troubleshooting
+
+The following table summarizes common issues customers may encounter when using external encryption keys and recommended actions:
+
+| Issue                             | Possible cause                                                   | Resolution                                                                                                                                                                                                                         |
+| --------------------------------- | ---------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Cluster cannot access KMS key     | Key policy does not grant the Camunda Cluster Amazon Role access | Update KMS key policy to include the correct Amazon Role ARN from the Camunda Console                                                                                                                                              |
```

Comment:
Cluster cannot access KMS key - is there an error message that customer can see? What would be the issue that customers face?